client/packages/common/src/hooks/useNativeClient/useNativeClient.ts

```diff
@@ -139,7 +93,7 @@ export const useNativeClient = ({
 
     let connectToPreviousTimer: NodeJS.Timer | undefined = undefined;
 
-    if (autoconnect) {
+    if (autoconnect && !!state.previousServer) {
```

Comment:
Great ! I also noticed this and had similar idea for fix, but that was when localStorage was `sync` method, not sure if this would always work now, since state.previousServer may not be set yet due to async nature of getting previousServer